gui/src/ScenarioVisualizer.tsx

```diff
@@ -0,0 +1,652 @@
+/*
+ * Copyright 2023 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import React, { useState, useCallback } from "react";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import Grid from "@mui/material/Grid";
+import IconButton from "@mui/material/IconButton";
+import Dialog from "@mui/material/Dialog";
+import TextField from "@mui/material/TextField";
+import Typography from "@mui/material/Typography";
+import {
+  FlagOutlined,
+  Close,
+  DeleteOutline,
+  Add,
+  Send,
+  CheckCircle,
+  Cancel,
+  ArrowForwardIosSharp,
+} from "@mui/icons-material";
+
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  AccordionSummaryProps,
+  Autocomplete,
+  Chip,
+  Divider,
+  styled,
+} from "@mui/material";
+import { Cycles, Property, Scenarios } from "./utils";
+
+interface ScenarioVisualizerProps {
```

Comment:
I guess the component will receive the scenario as one property of type ScenarioFull